### DISCUSSION TITLE
Modifying `+layout.server.js` (caching) responses headers

### DISCUSSION THREAD

**Discussion by kael** [sveltejs/kit#10723] [Q&A]
Is it possible to modify layouts responses headers ?

I'd like to set caching headers (`etag`) on some `+layout.server.js` that return data that take some time to be processed.

Testing with the following code:

```js
// +layout.server.js

import { fetchData } from '$api/data';

export function load({ url, setHeaders }) {
  const id = url.searchParams.get('id')
  setHeaders({ etag: "bar" });
  return {
    resource: fetchData(id),
  };
}
```

the `etag` is not set to layout headers but to the main requested url ones.

Is there some way to add caching headers to `+server.layout.js` ?

**Comment by stephane-vanraes** [comment]
> the `etag` is not set to layout headers but to the main requested url ones.

layouts are not pages, they cannot have their own headers, the `setHeaders` here will set the headers on the requested page
